### PR TITLE
Fill Transform to round to the current minute

### PR DESCRIPTION
**Pull request by studanshu**

### CODE REVIEW

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/metric/transform/FillTransform.java [hunk 1]
```diff
@@ -146,19 +148,9 @@ private static long _parseTimeIntervalInSeconds(String interval) {
         long offsetInSeconds = _parseTimeIntervalInSeconds(constants.get(3));
         double value = Double.parseDouble(constants.get(4));
 
-        SystemAssert.requireArgument(startTimestamp < endTimestamp, "End time must occure later than start time!");
+        SystemAssert.requireArgument(startTimestamp < endTimestamp, "End time must occur later than start time!");
         SystemAssert.requireArgument(windowSizeInSeconds >= 0, "Window size must be greater than ZERO!");
 
-        boolean isDivisible = ((startTimestamp - endTimestamp) % (windowSizeInSeconds * 1000)) == 0;
```
Comment (sundeepsf) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/metric/transform/FillTransform.java]: we still need this snapping logic if the window is not divisible by the step size

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/metric/transform/FillTransform.java [hunk 2]
```diff
@@ -173,6 +180,12 @@ private static long _parseTimeIntervalInSeconds(String interval) {
         return lineMetrics;
     }
 
+    private long roundOffToNearestMinute(long timeInMillis) {
```
Comment (sundeepsf) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/metric/transform/FillTransform.java]: minor, this is not nearest minute, but the current minute

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/metric/transform/FillTransform.java [hunk 3]
```diff
@@ -181,20 +180,30 @@ private static long _parseTimeIntervalInSeconds(String interval) {
...
+                    retVal = Long.parseLong(timeStr);
+                }
+            } catch (NumberFormatException nfe) {
+                throw new SystemException("Could not parse time.", nfe);
             }
-            return Long.parseLong(timeStr);
-        } catch (NumberFormatException nfe) {
-            throw new SystemException("Could not parse time.", nfe);
         }
+        return roundOffToNearestMinute(retVal);
```
Comment (sundeepsf) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/metric/transform/FillTransform.java]: actually thinking a bit more, can we do this only if the step size is more than a minute

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/metric/transform/FillTransform.java (hunk 2 above)
Comment (studanshu) [reply ArgusCore/src/main/java/com/salesforce/dva/argus/service/metric/transform/FillTransform.java]: @sundeepsf will make changes.

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/metric/transform/FillTransform.java (hunk 3 above)
Comment (studanshu) [reply ArgusCore/src/main/java/com/salesforce/dva/argus/service/metric/transform/FillTransform.java]: ok.

**Pull request merged**